File: lavcode-node/src/actions/folder/dtos/get-folder.dto.ts
import { FolderEntity } from "../../../entities/folder.entity";
import { IconEntity } from "../../../entities/icon.entity";
import { GetIconDto } from "../../../dtos/get-icon.dto";

export class GetFolderDto {
  id!: string;
  name!: string;
  order!: number;
  updatedAt!: number;

  icon?: GetIconDto;

  public static fromEntity(entity: FolderEntity, icon?: IconEntity) {
    const result = new GetFolderDto();
    result.id = entity._id;
    result.name = entity.name;
    result.order = entity.order;
    result.updatedAt = entity.updatedAt;
    result.icon = icon ? GetIconDto.fromEntity(icon) : undefined;
    return result;
  }
}

File: Lavcode.Service.Api/Dtos/GetFolderDto.cs
using Lavcode.Model;
using System;

namespace Lavcode.Service.Api.Dtos
{
    public class GetFolderDto
    {
        public string Id { get; set; }
        public string Name { get; set; }
        public int Order { get; set; }
        public DateTime LastEditTime { get; set; }

        public GetIconDto Icon { get; set; }



        public FolderModel ToModel()
        {
            return new FolderModel()
            {
                Id = Id,
                Name = Name,
                Order = Order,
                LastEditTime = LastEditTime,
            };
        }
    }
}
